src/rules.jl

```diff
@@ -161,11 +161,10 @@ hasdiffrule(f::Symbol, arity::Int) = in((f, arity), DEFINED_DIFFRULES)
 @define_diffrule /(x, y) = :(inv($y)),            :(-$x / ($y^2))
 @define_diffrule ^(x, y) = :($y * ($x^($y - 1))), :(($x^$y) * log($x))
 
-# TODO:
-#
-# mod
-# hypot
-# atan2
+@define_diffrule atan2(x, y) = :($y / ($x*$x + $y*$y)), :(-$x / ($x*$x + $y*$y))
+@define_diffrule hypot(x, y) = :($x / hypot($x, $y)), :($y / hypot($x, $y))
+@define_diffrule mod(x, y)   = :(1), :(-trunc($x / $y))
+@define_diffrule rem(x, y)   = :(1), :(-trunc($x / $y))
```

Comment:
Nope. I was wrong about being wrong. :smile: The rule works for `rem`, but `trunc` needs to be changed to `floor` for `mod`.